https://issues.redhat.com/browse/ACM-8498 Default alerts thanos

dockerymick opened the pull request:
https://issues.redhat.com/browse/ACM-8498

In the diagram, we have Thanos infrastructure. In the issue, Thanos compactor is mentioned. Are the infrastructure and compactor the same? Is the compactor a part of the infrastructure?

dockerymick committed:
https://issues.redhat.com/browse/ACM-8498 Default alerts thanos

dockerymick committed:
Other improvements

dockerymick commented on release_notes/whats_new.adoc:
```diff
@@ -56,6 +56,9 @@ For other Application topics, see link:../applications/app_management_overview.a
 * Use the _Fleet view_ switch from the _Overview_ page header for your Grafana metrics to be displayed, and filter the page data by using cluster labels. See, link:../console/console.adoc#console-overview[Console overview] for more details.
 
+//2.9:7362
+* The following alerts are now the default alerts for the Thanos infrastructure: `ACMThanosCompactHalted`, `ACMThanosCompactHighCompactionFailures`, `ACMThanosCompactBucketHighOperationFailures`, `ACMThanosCompactHasNotRun`. See link:../observability/observability_arch.adoc#observability-arch[Observability architecture].
```
```suggestion
* The following default alerts are provided for the Thanos compactor: `ACMThanosCompactHalted`, `ACMThanosCompactHighCompactionFailures`, `ACMThanosCompactBucketHighOperationFailures`, `ACMThanosCompactHasNotRun`. See link:../observability/observability_arch.adoc#observability-arch[Observability architecture].
```

dockerymick committed:
Update release_notes/whats_new.adoc

douglascamata commented:
> In the diagram, we have Thanos infrastructure. In the issue, Thanos compactor is mentioned. Are the infrastructure and compactor the same? Is the compactor a part of the infrastructure?

That's correct, @dockerymick. Thanos Compactor is part of the Thanos infrastructure. To put it in the language used in the Thanos project, Compactor is one of the components in Thanos. 👍

douglascamata commented on observability/observability_arch.adoc:
```diff
@@ -15,6 +15,31 @@ The components of the observability architecture include the following items:
 
 - The _observability add-on controller_ is the API server that automatically updates the log of the managed cluster.
 
+- The Thanos infrastructure, includes the Thanos compactor, which is deployed by the `multicluster-observability-operator` pod. The Thanos compactor ensures that queries are performing well by using the retention configuration, and compaction of the data in storage.
```
I suggest to capitalize `compactor` here. In official Thanos documentation, `Compactor` is always referred to like so. It's the proper name of the component.

dockerymick committed:
Update observability_arch.adoc

Updating Thanos Compactor reference

swopebe commented on observability/observability_arch.adoc:
```diff
@@ -15,6 +15,31 @@ The components of the observability architecture include the following items:
 - The _observability add-on controller_ is the API server that automatically updates the log of the managed cluster.
 
+- The Thanos infrastructure includes the Thanos Compactor, which is deployed by the `multicluster-observability-operator` pod. The Thanos Compactor ensures that queries are performing well by using the retention configuration, and compaction of the data in storage.
++
```
Beautiful description!

swopebe commented on observability/observability_arch.adoc:
```diff
@@ -15,6 +15,31 @@ The components of the observability architecture include the following items:
+|===
+| Alert | Severity | Description
+
+| ACMThanosCompactHalted
```
These should be coded: `ACMThanosCompactHalted`

this will also help with translation tools, as they won't be scanned as actual words that people have to translate.

swopebe committed:
Update observability_arch.adoc

changed for this table, you can decide if you have time to change in other spots where things should be coded.

swopebe reviewed (approved):
made some small changes.

swopebe commented on observability/observability_arch.adoc:
```diff
@@ -15,6 +15,31 @@ The components of the observability architecture include the following items:
+
+| `ACMThanosCompactHighCompactionFailures`
+| warning
+| An alert is sent when the compaction failure rate is greater than 5%.
```
check to see if we should be writing out percent???

dockerymick replied:
I checked the style guide and there were no requirements mentioned. However, I agree with the update
```suggestion
| An alert is sent when the compaction failure rate is greater than 5 percent.
```

dockerymick committed:
Update observability/observability_arch.adoc

douglascamata reviewed (approved):
/lgtm

dockerymick commented:
@douglascamata @swopebe thanks for the reviews and approvals! Merging this PR
